File: androidp1/app/src/main/java/com/preappointment1/app/data/AuthHelper.kt
package com.preappointment1.app.data

import android.util.Log
import com.preappointment1.app.data.api.ApiClient
import com.preappointment1.app.data.model.AuthRequest

object AuthHelper {
    private const val TAG = "LPM_AUTH"
    private const val DEVICE_PASSWORD = "secret_device_password"

    suspend fun ensureAuthenticated(): Boolean {
        if (SessionManager.getToken() != null) return true

        val email = "${SessionManager.getOrCreateDeviceId()}@local.device"
        val request = AuthRequest(email = email, password = DEVICE_PASSWORD)

        return try {
            val response = ApiClient.apiService.register(request)
            SessionManager.saveToken(response.token)
            Log.d(TAG, "Registered new device account")
            true
        } catch (e: Exception) {
            try {
                val response = ApiClient.apiService.login(request)
                SessionManager.saveToken(response.token)
                Log.d(TAG, "Logged in existing device account")
                true
            } catch (loginError: Exception) {
                Log.e(TAG, "Authentication failed", loginError)
                false
            }
        }
    }

    suspend fun ensureProfile(): String? {
        SessionManager.getProfileId()?.let { return it }

        val profile = ApiClient.apiService.createProfile(
            com.preappointment1.app.data.model.ProfileRequest(
                first_name = "Patient",
                last_name = "Local",
                relation = "Self"
            )
        )
        SessionManager.saveProfileId(profile.id)
        return profile.id
    }
}

File: iosp1/P1/P1/Network/AuthHelper.swift
import Foundation

class AuthHelper {
    static let shared = AuthHelper()
    
    private let devicePassword = "secret_device_password"
    
    private init() {}
    
    func ensureAuthenticated() async -> Bool {
        if SessionManager.shared.getToken() != nil {
            return true
        }
        
        let email = "\(SessionManager.shared.getOrCreateDeviceId())@local.device"
        let request = AuthRequest(email: email, password: devicePassword)
        
        do {
            let response = try await ApiService.shared.register(request: request)
            SessionManager.shared.saveToken(response.token)
            print("LPM_AUTH: Registered new device account")
            return true
        } catch {
            do {
                let response = try await ApiService.shared.login(request: request)
                SessionManager.shared.saveToken(response.token)
                print("LPM_AUTH: Logged in existing device account")
                return true
            } catch {
                print("LPM_AUTH: Authentication failed - \(error)")
                return false
            }
        }
    }
    
    func ensureProfile() async -> String? {
        if let profileId = SessionManager.shared.getProfileId() {
            return profileId
        }
        
        do {
            let profile = try await ApiService.shared.createProfile(
                request: ProfileRequest(first_name: "Patient", last_name: "Local", relation: "Self")
            )
            SessionManager.shared.saveProfileId(profile.id)
            return profile.id
        } catch {
            print("LPM_AUTH: Failed to create profile - \(error)")
            return nil
        }
    }
}
